src/web/rustdoc.rs

```diff
@@ -1040,17 +980,43 @@ mod test {
     #[test]
     fn badges_are_urlencoded() {
         wrapper(|env| {
+            use reqwest::Url;
+            use url::Host;
+
             env.fake_release()
                 .name("zstd")
                 .version("0.5.1+zstd.1.4.4")
                 .create()?;
 
-            let frontend = env.frontend();
-            assert_redirect(
-                "/zstd/badge.svg",
-                "/zstd/badge.svg?version=0.5.1%2Bzstd.1.4.4",
-                &frontend,
-            )?;
+            let frontend = env.override_frontend(|frontend| {
+                use reqwest::blocking::Client;
+                use reqwest::redirect::Policy;
+                // avoid making network requests
+                frontend.client = Client::builder().redirect(Policy::none()).build().unwrap();
+            });
+            let mut last_url = "/zstd/badge.svg".to_owned();
+            let mut response = frontend.get(&last_url).send()?;
+            let mut current_url = response.url().clone();
+            // follow redirects until it actually goes out into the internet
+            while !matches!(current_url.host(), Some(Host::Domain(_))) {
```

Comment:
why can we have more than one redirect here? shouldn't `/zstd/badge.svg` directly return the redirect to shields.io?

Reply:
The test handler redirects from the relative path to an absolute HTTP scheme:
```
[2021-07-06T00:17:45Z INFO  docs_rs::web] Running docs.rs web server on http://127.0.0.1:40949
(/zstd/badge.svg -> http://127.0.0.1:40949/zstd/badge.svg)
(http://127.0.0.1:40949/zstd/badge.svg -> https://img.shields.io/docsrs/zstd/latest)
```